### ISSUE TITLE
Add check in github actions for see if jenkinsfile in PR has a test associated with it

### ISSUE THREAD

**Issue by abhinavGupta16** [opensearch-project/opensearch-build#1444]
### Is your feature request related to a problem? Please describe

We need to have regression tests for all jenkins job that we create. As the number of contributors for the jobs increase and the number of jobs increase, we need a check that would make sure every jenkinsfile that is added has a regression test for it, else, the ci would prevent the PR from merging.

### Describe the solution you'd like

Add a check to make sure all jenkinsfile have a test associated with it.

One possible solution could be to check if every `jenkinsfile` in the PS has a `jenkinsfile.txt` as well.

### Acceptance Criteria
- [ ] given I create a jenkins file without a test, ci should fail preventing me from merging the PR

**Comment by dblock** [comment]
Good idea. Note that sometimes it's not possible, notably because issues such as https://github.com/jenkinsci/JenkinsPipelineUnit/issues/434, but we should be able to at least have placeholders that reference these.

**Comment by dblock** [comment]
Could also think of this as code coverage and report Jenkinsfile's as code and see whether those are tested or not.

**Comment by gaiksaya** [comment]
[Triage] @abhinavGupta16 Can you please create issues for existing jenkinsfile?
Thanks!

**Comment by rblcoder** [comment]
@gaiksaya  Can this be a separate python file run in a separate GHA?
